test/wallet-http-test.js

```diff
@@ -86,6 +88,20 @@ describe('Wallet HTTP', function() {
     await node.mempool.reset();
   });
 
+  it('should get key by address from watch-only primary/foobar', async () => {
+    const accountInfo = await wallet.getAccount(accountTwo);
+    await wclient.createWallet('watchonly', {
+      watchOnly: true,
+      accountKey: accountInfo.accountKey
+    });
+    const key = await wallet3.getKey(accountInfo.receiveAddress);
+    const xpub = HD.HDPublicKey.fromBase58(accountInfo.accountKey, 'regtest');
+    assert.equal(accountInfo.accountIndex, key.account);
+    assert.equal(xpub.childIndex ^ HD.common.HARDENED, key.account);
+    assert.equal(0, key.branch);
+    assert.equal(0, key.index);
+  });
```

Comment:
I feel like this test should explicitly assert that `account.accountIndex != key.childIndex` since that's what we're really trying to solve. Since you're using an account generated by a previous test, regression is otherwise possible. You can use `wallet.getAccountIndex()`

Reply:
We discussed out of band and decided to derive an account key and address within the test. This allows the reader to be sure of the xpub's account index.